pkgs/coverage/lib/src/coverage_options.dart

```
@@ -0,0 +1,120 @@
+  });
+
+  factory CoverageOptions.fromConfig(
+      Config options, CoverageOptions defaultOptions, String? optionsFilePath) {
+    var outputDirectory = options.optionalString('output_directory') ??
+        defaultOptions.outputDirectory;
+    var packageDirectory = options.optionalString('package_directory') ??
+        defaultOptions.packageDirectory;
+
+    if (optionsFilePath != null) {
+      if (outputDirectory != null && !path.isAbsolute(outputDirectory)) {
+        outputDirectory = path.normalize(
+            path.absolute(path.dirname(optionsFilePath), outputDirectory));
+      }
+      if (!path.isAbsolute(packageDirectory)) {
+        packageDirectory = path.normalize(
+            path.absolute(path.dirname(optionsFilePath), packageDirectory));
+      }
+    }
+
+    return CoverageOptions(
+      outputDirectory: outputDirectory,
+      scopeOutput: options.optionalStringList('scope_output') ??
+          defaultOptions.scopeOutput,
+      functionCoverage: options.optionalBool('function_coverage') ??
+          defaultOptions.functionCoverage,
+      branchCoverage: options.optionalBool('branch_coverage') ??
+          defaultOptions.branchCoverage,
+      packageDirectory: packageDirectory,
+      packageName:
+          options.optionalString('package_name') ?? defaultOptions.packageName,
+      testScript:
+          options.optionalString('test_script') ?? defaultOptions.testScript,
+    );
+  }
+
+  final String? outputDirectory;
+  final List<String> scopeOutput;
+  final bool functionCoverage;
+  final bool branchCoverage;
+  final String packageDirectory;
+  final String? packageName;
+  final String testScript;
+}
+
+class CoverageOptionsProvider {
+  CoverageOptionsProvider({
+    String? filePath,
+  }) {
+    final file = _getOptionsFile(filePath);
+    final fileContents = file?.readAsStringSync();
+
+    final isFileEmpty = fileContents?.isEmpty ?? false;
+
+    // Pass null to fileContents if the file is empty
```

Just a question: Why do you need to pass null to `fileContents` if the file is empty?

---

If we pass empty file content to `fromConfigFileContents`, it throws a `FormatException`. Should we also handle cases where the file is empty but contains only blank spaces?

---

No, you can let the exception be thrown so the user can see it. You could even let that happen for empty files.